### ISSUE TITLE
provide a complete for template syntax?

### ISSUE THREAD

**Issue by Freed-Wu** [aperezdc/vim-template#140]
just like lsp, when user write a language will auto prompt the keyword of this language?

![image](https://user-images.githubusercontent.com/32936898/79970737-f611b200-84c5-11ea-9c69-9de51d81498a.png)

**Comment by fimmind** [comment]
Now I get such an error whenever I startup neovim:
![04 06 20_22:38:07:204643299](https://user-images.githubusercontent.com/44144507/83802819-1c756080-a6b4-11ea-819c-ae0eefa399db.png)

[There](https://github.com/fimmind/Dotfiles/tree/master/nvim) is my nvim config, if it's helpful for fixing

**Comment by Freed-Wu** [comment]
it is my error. function coc#source#template#foo() should be declared in autoload/coc/source/template.vim, but after test, i create a new file vim-template.vim. it should rename it to template.vim.

**Comment by Freed-Wu** [comment]
i found coc.nvim has a plugin <https://github.com/voldikss/coc-template> to realize the same function. amazing.

**Comment by Freed-Wu** [comment]
@fimmind Can <https://github.com/aperezdc/vim-template/pull/143> solve your problem?

**Comment by fimmind** [comment]
I'll try it within the day and tell you if it fixed the problem. (Sorry for such a late answer)

**Comment by Freed-Wu** [comment]
it is my pleasure!

**Comment by fimmind** [comment]
I tried #143 and now there is no error message on sturtup, but completion still doesn't work. I guess that's cause the plugin doesn't provide a way for vim to identify template files (e.g. `echo &ft` for `=template=.py` shows `python`, and completion works the same way as for any other Python file). I also tried adding 
```
exec 'autocmd BufEnter =template=* setlocal ft=template'
exec 'autocmd BufEnter .vim-template:* setlocal ft=template'
```
into `ftdetect/template.vim`, but this didn't help

**Comment by fimmind** [comment]
I figured out that I used a wrong filetype. Now I added following into `ftdetect/vim-template.vim` and everything works very well:
```
autocmd BufEnter =template=* setlocal ft=vim-template
autocmd BufEnter .vim-template:* setlocal ft=vim-template
```

But I looked at `vim-template`'s source code and found out that there are [lines of code](https://github.com/Freed-Wu/vim-template/blob/master/plugin/templates.vim#L526) attempting to set filetype similarly to how I did this, but somewhy they have no effect. Don't you now why could this happend?
